src/main/java/com/northconcepts/datapipeline/foundations/examples/pipeline/FilterColumnsWithAllNullValues.java

```diff
@@ -28,7 +30,7 @@ public static void main(String[] args) {
             .setOutput(pipelineOutput);
 
         Dataset dataset = new MemoryDataset(pipeline);
-        dataset.load().waitForRecordsToLoad();
+        dataset.load().waitForRecordsToLoad(100, TimeUnit.SECONDS.toMillis(4));
```

Comment:
@sah2020 I believe there is no issue with `waitForRecordsToLoad()`. Keep this line as it is.
Can you please add this line after the current line?
`dataset.waitForColumnStatsToLoad();`

The reason: the columns are loading in a separate thread. So even before columns are loaded, the code execution is completed and that could be the reason that the file is empty. It depends on the CPU and available resources on the machine for columns to load.